firebase-mirror/src/main/java/io/spine/server/firebase/FirestoreSubscriptionPublisher.java

```diff
@@ -23,34 +23,28 @@
 import com.google.cloud.firestore.CollectionReference;
 import com.google.cloud.firestore.DocumentReference;
 import com.google.cloud.firestore.WriteBatch;
-import com.google.common.annotations.VisibleForTesting;
-import com.google.common.collect.ImmutableMap;
-import com.google.protobuf.Any;
-import com.google.protobuf.Message;
-import io.spine.base.Identifier;
-import io.spine.client.EntityStateUpdate;
 import io.spine.logging.Logging;
-import io.spine.server.storage.StorageField;
-import org.slf4j.Logger;
 
 import java.util.Map;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.Future;
 
-import static com.google.cloud.firestore.Blob.fromBytes;
 import static com.google.common.base.Preconditions.checkNotNull;
-import static io.spine.protobuf.AnyPacker.unpack;
-import static io.spine.server.firebase.FirestoreSubscriptionPublisher.EntityStateField.bytes;
-import static io.spine.server.firebase.FirestoreSubscriptionPublisher.EntityStateField.id;
 
 /**
- * Publishes {@link EntityStateUpdate}s to <a target="_blank"
+ * Publishes {@link io.spine.client.SubscriptionUpdate}s to <a target="_blank"
  * href="https://firebase.google.com/docs/firestore/">Cloud Firestore^</a>.
+ *
+ * <p>The descendants elaborate on how to compose the document identifier and body for this
+ * particular subscription.
+ *
+ * @param <U>
+ *         the subscription update payload format
  */
-final class FirestoreSubscriptionPublisher {
+abstract class FirestoreSubscriptionPublisher<U> implements Logging {
```

Comment:
Isn't it just `FirestorePublisher`? And `U` seems to be just an "update" we publish.